### ISSUE TITLE
80080204: App manifest validation error during building of XamlIsland apps

### ISSUE THREAD

**Issue by michalleptuch** [CommunityToolkit/Microsoft.Toolkit.Win32#328]
## Describe the bug
I can't build my XamlIsland apps in Visual Studio 2019 v16.10 and higher. Even Visual Studio 2022 Preview doesn't handle it. Visual Studio 2019 v16.9.8 is the last tested and working version.
```
Error	APPX0501	Validation error. error 80080204: App manifest validation error: Line 66, Column 27, Reason: The field "*[local-name()='Extensions']/*[local-name()='Extension']/*[local-name()='InProcessServer']/*[local-name()='ActivatableClass']" with value "Microsoft.Toolkit.Win32.UI.XamlHost.XamlApplication" must only be declared once. A duplicate exists on Line 60, Column 27.
```
## Steps to Reproduce
1. Clone this repo: https://github.com/michalleptuch/rounded-flyout
2. Open in VS and build
3. See error in Output window

## Expected behavior
Correctly building of solution

## Environment
```
NuGet Package(s): 

Package Version(s): 

Project .NET Version:
-  .NET Core 3.1

Packages:
- Microsoft.Toolkit.Win32.UI.XamlApplication (6.0.0-6.1.3)

Windows 10 Build Number:
- 18362-22000

App min and target version:
- 18362-22000

Device form factor:
- Desktop

Visual Studio 
- 2019 (version: 16.10.x) 
- 2022 Preview (version: 17.0)

```

## Additional context
I found similar issues that describe the same situations but solutions don't work for me:
- https://stackoverflow.com/questions/64791579/app-manifest-validation-error-after-update-to-visual-studio-2019-16-8-0
- https://developercommunity.visualstudio.com/t/Error-when-building-UWP-app-with-packagi/1258250#T-ND1273557

**Comment by michael-hawker** [comment]
FYI @marb2000 @ocalvo

**Comment by filiphsandstrom** [comment]
We also experience this on https://github.com/w10m-research/AdaptiveShell/commit/d0631d65b335773c72f89345594c27c9aa22da3d

**Comment by Samuel12321** [comment]
The ModernFlyouts Team is also experiencing this issue.
This is a major breaking bug that has had no solution for 4 months.
@marb2000 @ocalvo @michael-hawker

**Comment by atiyka** [comment]
I also experienced the same issue when I updated the VisualStudio2019. As a temporal solution I had to go back to the previous version and I can't upgrade until this bug won't be fixed.

**Comment by michael-hawker** [comment]
Has anyone filed a new issue with the Visual Studio team for this regression? If so, can they link to that issue here?

**Comment by Samuel12321** [comment]
Has been reported here https://developercommunity2.visualstudio.com/t/Visual-Studio-16100-breaks-support-for/1634426?q=microsoft.toolkit&ftype=problem&space=8&stateGroup=active&sort=newest

**Comment by Samuel12321** [comment]
> Has been reported here https://developercommunity2.visualstudio.com/t/Visual-Studio-16100-breaks-support-for/1634426?q=microsoft.toolkit&ftype=problem&space=8&stateGroup=active&sort=newest

Can everyone please upvote this issue so it can get fixed.

**Comment by ocalvo** [comment]
It seems that the toolkit [msbuild code](https://github.com/CommunityToolkit/Microsoft.Toolkit.Win32/blob/master/Microsoft.WinRT.Win32.targets#L138) that detects if the Xaml Applicaton dll has been added already is now not detecting it and ends up adding 2 entries of the same dll.

**Comment by Samuel12321** [comment]
Seems to work, Thanks @ocalvo.
When should we expect this to be properly fixed?

**Comment by michalleptuch** [comment]
@ocalvo Yes, it works for building, but during packaging app for the Store I get this error:
```
1>PRI175: 0x80073b0f - Processing Resources failed with error: Duplicate Entry.
1>PRI222: 0x80073b0f - Unspecified error occurred.
```
Branch: [FixForVS](https://github.com/michalleptuch/rounded-flyout/tree/FixForVS)

**Comment by atiyka** [comment]
Thank you @ocalvo, it works for me as well

**Comment by atiyka** [comment]
@ocalvo, after I added your suggestion to the project, I submitted the build to the Store, but the certification process did not start at all. After 4 days of waiting we received an email from the support:
`The submission has been cancelled due to an invalid package. Error details: Error: DirectWriteForwarder : Gatekeeper error GK0028 : GK0028 Assembly ‘DirectWriteForwarder’ is a mixed mode assembly containing managed and native code.  Only pure managed assemblies are compatible with .NET Native.`
I reverted this change, created an app package with an older VS 2019 (16.9.1) and tried to submit to the store. Almost immediatelly started the certification process, so I think this workaround cause the issue.

Do you have any idea?

**Comment by atiyka** [comment]
Well, I just added the code above (except the first 4 lines) at the end of the `.wapproj` file in the UWP project before the closing `</Project>` tag. It created the app package without any issues for the packaging project in VS.

**Comment by Samuel12321** [comment]
This is a major breaking bug that has been a problem for 6 months now. Any eta on when we can expect an official solution / patch shipped?
@marb2000 @ocalvo @michael-hawker

**Comment by aquinn39** [comment]
Also reported here: https://github.com/MicrosoftDocs/windows-dev-docs/issues/3544. This issue still exists in the latest version of Visual Studio 2022 unfortunately so I'm still stuck using Visual Studio 2019 16.9.8.
